kalliope/core/RestAPI/FlaskAPI.py

```diff
@@ -215,30 +221,48 @@ def run_synapse_by_audio(self):
                 "error": "No file provided"
             }
             return jsonify(error=data), 400
-        if file and self.allowed_file(file.filename):
-            # save the file
-            filename = secure_filename(file.filename)
-            base_path = os.path.join(self.app.config['UPLOAD_FOLDER'])
-            file.save(os.path.join(base_path, filename))
-
-            # now start analyse the audio with STT engine
-            audio_path = base_path + os.sep + filename
-            ol = OrderListener(callback=self.audio_analyser_callback, audio_file_path=audio_path)
-            ol.start()
-            ol.join()
-            # wait the Order Analyser processing. We need to wait in this thread to keep the context
-            while not self.order_analyser_return:
-                time.sleep(0.1)
-            self.order_analyser_return = False
-            if self.api_response is not None and self.api_response:
-                data = jsonify(self.api_response)
-                self.api_response = None
-                return data, 201
-            else:
-                data = {
-                    "error": "The given order doesn't match any synapses"
-                }
-                return jsonify(error=data), 400
+        # save the file
+        filename = secure_filename(file.filename)
+        base_path = os.path.join(self.app.config['UPLOAD_FOLDER'])
+        file.save(os.path.join(base_path, filename))
+
+        # now start analyse the audio with STT engine
+        audio_path = base_path + os.sep + filename
+        logger.debug("[FlaskAPI] run_synapse_by_audio: with file path %s" % audio_path)
+        if not self.allowed_file(audio_path):
+            audio_path = self._convert_to_wav(audio_file_path=audio_path)
+        ol = OrderListener(callback=self.audio_analyser_callback, audio_file_path=audio_path)
+        ol.start()
+        ol.join()
+        # wait the Order Analyser processing. We need to wait in this thread to keep the context
+        while not self.order_analyser_return:
+            time.sleep(0.1)
+        self.order_analyser_return = False
+        if self.api_response is not None and self.api_response:
+            data = jsonify(self.api_response)
+            self.api_response = None
+            logger.debug("[FlaskAPI] run_synapse_by_audio: data %s" % data)
+            return data, 201
+        else:
+            data = {
+                "error": "The given order doesn't match any synapses"
+            }
+            return jsonify(error=data), 400
+
+    @staticmethod
+    def _convert_to_wav(audio_file_path):
```

Comment:
This method should test the extension before. In order to escape the conversion if we are already using wav.
Don't you think?